crates/merkle-tree/src/contract.rs

```diff
@@ -66,7 +67,7 @@ impl<'tx> ContractsStorageTree<'tx> {
             block: Some(block),
             contract,
         };
-        let tree = MerkleTree::new(root);
+        let tree = MerkleTree::new(TrieStorageIndex::new(root));
```

Comment:
Looks like `tx.contract_root_index()` is still returning a trie storage index as an u64. It should be modified to return a `Result<Option<TrieStorageIndex>>` like `class_root_index()` and `storage_root_index()`.